stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -527,13 +527,15 @@ end
 eigvals(D::Diagonal{<:Number}; permute::Bool=true, scale::Bool=true) = D.diag
 eigvals(D::Diagonal; permute::Bool=true, scale::Bool=true) =
     [eigvals(x) for x in D.diag] #For block matrices, etc.
```

Comment:
this is wrong, I think: for a block matrix it will give an array of arrays rather than an array of scalars.

Reply:
I wrote it that way because the i-th element of the eigvals matrix will correspond to the i-th element of the eigvecs matrix making it a eigen value - eigen vector pair for i-th diagonal matrix in the given block diagonal matrix.

If that is a mistake, could you suggest the necessary correction required

Reply:
Something like
```
eigvals(D::Diagonal; permute::Bool=true, scale::Bool=true) = vcat([eigvals(d) for d in D.diag]...)
```
This one has at least type promotion, in contrast to the definition above, which may yield a mixed vector `5-element Array{Array{T,2} where T,1}`, with real and complex array elements.